lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -345,6 +345,20 @@ rewriteAddWithNegativeConstant(comb::AddOp add, hw::ConstantOp rhsCst,
   return negCst;
 }
 
+// Transforms a hw.struct_explode operation into a set of hw.struct_extract
+// operations.
+static void lowerStructExplodeOp(hw::StructExplodeOp op) {
+  ImplicitLocOpBuilder builder(op.getLoc(), op);
+  StructType structType = op.getInput().getType().cast<StructType>();
+  for (auto [res, field] :
+       llvm::zip(op.getResults(), structType.getElements())) {
+    auto extract =
+        builder.create<hw::StructExtractOp>(op.getInput(), field.name);
+    res.replaceAllUsesWith(extract);
+  }
+  op.erase();
+}
+
```

Comment:
Can we return a first operation created in the function and set `opIterator` to the returned operation as `rewriteAddWithNegativeConstant` does?